src/main/java/com/eerussianguy/blazemap/mixin/ServerLevelMixin.java

```diff
@@ -1,26 +1,44 @@
 package com.eerussianguy.blazemap.mixin;
 
+import java.util.List;
+import java.util.concurrent.Executor;
+
 import net.minecraft.core.BlockPos;
+import net.minecraft.core.Holder;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.progress.ChunkProgressListener;
+import net.minecraft.world.level.Level;
 import net.minecraft.world.level.ChunkPos;
+import net.minecraft.world.level.CustomSpawner;
 import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.chunk.ChunkGenerator;
+import net.minecraft.world.level.dimension.DimensionType;
+import net.minecraft.world.level.storage.LevelStorageSource;
+import net.minecraft.world.level.storage.ServerLevelData;
 
 import com.eerussianguy.blazemap.engine.server.BlazeMapServerEngine;
+
 import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 @Mixin(ServerLevel.class)
-public abstract class ServerLevelMixin {
+public abstract class ServerLevelMixin extends Level {
```

Comment:
Why `Level` instead of `ServerLevel`, since it's the class we're targeting?

Reply:
Short answer: Because that's what the docs told me to do. (https://github.com/SpongePowered/Mixin/wiki/Introduction-to-Mixins---Understanding-Mixin-Architecture)

Long answer: Because this class doesn't really exist. It will cease to exist when first loaded at runtime as Mixin merges the bytecode from the mixin with the base class. Therefore, any methods added to this class must behave as if they were operating within the scope of `ServerLevel`. As `ServerLevel extends Level`, so to does `ServerLevelMixin extends Level`. 

Also it gives me access to `.dimension()`, which is defined on the super class, without giving me compilation errors 😛

Reply:
This was mostly related to the super vs this argument. I understand the class doesn't exist at runtime and it's just a container for bytecode that will be injected elsewhere... My concern is more about generating bytecode that is as close to as if it was code from the class itself rather than injected as possible.

But to be honest, given how much we fought to get here, as long as it works I won't be mad.
This isn't about nitpicking 😄 maybe more on the side of trying to leave this squeaky clean on the first attempt to that we can stop messing with Mixins and do other stuff :)